### ISSUE TITLE
Is there an official Oracle Database on Windows Container image? 

### ISSUE THREAD

**Issue by sebagomez** [oracle/docker-images#494]
The subject is my question... if it's not, is there a restriction why it couldn't be done?

**Comment by brunoborges** [comment]
There is not, but you should be able to build one based out of the files we have here. Want to give it a try?

**Comment by grahamehorner** [comment]
@sebagomez   @brunoborges would be good to have an official Windows server core image, but I'm happy to give it ago

**Comment by sebagomez** [comment]
I'll try to work on that too.
If you want share a repo to it and I'll PR to help out in anyway I can

**Comment by grahamehorner** [comment]
is there an official installer for windows other than the setup.exe or an installer which can be controlled via a script so not to require user interface or user input ?

**Comment by gvenzl** [comment]
Yes, you can use response files to perform a silent install. Have a look at [Installing and Configuring Oracle Database Using Response Files](http://docs.oracle.com/database/122/NTDBI/advance.htm#NTDBI009) for more information.

**Comment by grahamehorner** [comment]
@gvenzl thanks I've forked this repo and will create a version with a Windows Server Core OracleDatabase docker build; once I'm happy I'll then submit a pull request for a review.

**Comment by mawl** [comment]
I would like to install the oracle odbc driver from **winnt_12c_client32** in a windows docker image - but the silent installation doesn't work in a windowsservercore image with no further output. In my opion the installation isn't silent enough, still wants to open a window or something. Any ideas?

**Comment by grahamehorner** [comment]
@mawl  I came across an issue with the install when I hadn't told windows to trust the download before extracting the .zip file :'(  took 3days to find out; personally I've switched to using the Oracle Managed DataAccess Components for .NET to avoid client installation of drivers etc. that use the system registry & GAC

I realise this response is .net related but hope the info helps

**Comment by srirams007** [comment]
I also tried to install the Oracle Client 12c on the windows server core container image. Looks setup.exe completed without doing any actions. No logs or no errors on the console. Is Anyone tried ?.

**Comment by GillIntergraph** [comment]
Same here

**Comment by Djelibeybi** [comment]
Guys, please open a new issue for this,.

**Comment by JuanPerezActivus** [comment]
I have been struggling with this for a couple of days

Oracle setup installer and response files ready, I run the setup command with my response file on my own PC from an elevated cmd window and Oracle is installed, no problems found and oracledb installs according to my reponsefile

CMD command
setup -silent -responsefile d:\temp\database\response\db.rsp -skipPrereqs -noconfig -waitForCompletion

But even with silent switch the Oracle Universal Installer and some configuration still popup, that will not work when installing from an image or within a container itself
(see picture attached)

![](https://discourse-cdn-sjc1.com/docker/uploads/default/original/3X/8/7/87af8cc17388e5294946fb0f60b692ce77543cb0.png)


When try to replicate using my dockerfile or an error or nothing happens

Docker file
#Sample Dockerfile

#Indicates that the windowsservercore image will be used as the base image.
FROM microsoft/windowsservercore

#Create the working folder
RUN mkdir c:\temp\database

#Copy Oracle installers and response file to the working directory previously created
COPY database /temp/database

#Run the command
RUN Powershell.exe Start-Process -FilePath ‘C:\temp\database\setup.exe’ -argumentlist ‘-responsefile c:\temp\database\response\db.rsp’, ‘silent’ -wait

Once ready I run on an elevated powershell Window

PS D:\docker\dockerfiles> docker image build -t 2016oracle -f dockerfile .

As I said even that the build might finish successfully and without errors the application it is not installed or the container run for a moment and then exits few seconds later

**Comment by dkorel-copperleaf** [comment]
I gave up on this when i realized the java based installer was spawning numerous new windows during the silent install process. That, and the fact that it's going to try to write to the registry and i don't know if the registry is available in windows docker images.

I think if you can run a linux container on windows on the same system, then using a linux based Oracle container might be a better path... but i'm not sure if there is full support / network connectivity between LCOW and windows containers... yet. But i could bet that there will be support for that before Oracle installs into a windows container.

**Comment by orest-gulman** [comment]
hi guys, does anyone tryed this image?:
https://hub.docker.com/r/jturco/oraclecore

**Comment by yanko** [comment]
no update on this?

**Comment by JuanPerezActivus** [comment]
**_> no update on this?_**
Hi Yanko

I tried this official guide and other approaches but not vail, could not make it work if you do please share your outcome
**Running Oracle Database and
Applications in Docker Containers
on Windows**
https://www.oracle.com/technetwork/topics/dotnet/tech-info/oow18windowscontainers-5212844.pdf

**Comment by JuanPerezActivus** [comment]
> JuanPerezActivus,
> 
> Nice work! I know the frustration with some of those roadblocks...
> 
> I wonder if perhaps using a "state difference / snapshot" type tool (like an installer creator) to reverse engineer the installer?
> i.e:
> 
> * Perform a system snapshot before the install (on a regular system)
> * Perform the install
> * Export the system state differences (registry entries, file system changes)
> * Convert those changes to docker RUN layers
> 
> I realize this process would not allow for variation / customization in the tablespaces... perhaps once the database engine has been installed, dbca could be run as an entrypoint script to create, once, the tablespaces... Just an idea... might work if dbca runs without interruption...

Tried different approaches, always got install permissions issues which block the whole project

**Comment by iTeK02** [comment]
I'm encountering the same error.

`[FATAL] PRVG-1901 : failed to setup CVU remote execution framework directory "C:\Users\ContainerAdministrator\AppData\Local\Temp\InstallActions2020-07-17_01-18-31PM\CVU_18.0.0.0.0_ContainerAdministrator\" on nodes "975ef812cf39"
Please select a different work area for the framework
975ef812cf39 : PRKN-1014 : Failed to execute remote command "C:\Users\ContainerAdministrator\AppData\Local\Temp\InstallActions2020-07-17_01-18-31PM\CVU_18.0.0.0.0_ContainerAdministrator\\exectask.exe" on node "975ef812cf39".Failed during connecting to service
975ef812cf39 : Failed during connecting to service`

 No one's come up with a solution?

**Comment by civortech** [comment]
I ended up moving to WSL2 :(

**Comment by Sublime1** [comment]
6 years later since this issue was created, and still no way to make a Windows container for Oracle client. I tried [this guy's](https://github.com/ebellumat/docker-files-windows/tree/master/.Net%204.7.2%20%2B%20Oracle%20Data%20Access%20Components%20(ODAC)%2032%20bit) approach but it's giving a bunch of errors. This is incredible.

**Comment by vrapolinario** [comment]
I'm not an Oracle expert, but has anyone in this thread tried the Windows or Server images, rather than the Server Core one? The API surface on the Server Core image is limited, so I'd give it a try with the Server image. `mcr.microsoft.com/windows/server:ltsc2022`

**Comment by orest-gulman** [comment]
I've installed Oracle 19 on server 2019 and 2022 core using this instruction:
https://community.oracle.com/tech/developers/discussion/comment/16806105#Comment_16806105

**Comment by christianshay** [comment]
> I've installed Oracle 19 on server 2019 and 2022 core using this instruction: https://community.oracle.com/tech/developers/discussion/comment/16806105#Comment_16806105

Thanks for posting that which points to this presentation:
https://www.oracle.com/docs/tech/oow19-dockerforwindows.pdf

The key point is to use Oracle Database 19.7 or later to get the bug fixes pertinent to Windows containers and silent installs  and to use Windows Server Core, ltsc2019 image:

mcr.microsoft.com/windows/servercore:ltsc2019

For what it is worth, SQL Server support for Windows Containers was suspended in 2021:

https://techcommunity.microsoft.com/t5/sql-server-blog/update-beta-program-for-sql-server-on-windows-container-is/bc-p/2564521

I

**Comment by Sublime1** [comment]
@orest-gulman and @christianshay thank you both so much for this. I can confirm that I have the Oracle client running on a Windows Docker container. This is really great news. I used instantclient_18_5 as instantclient_18_3 is no longer available for download. https://www.oracle.com/database/technologies/instant-client/winx64-64-downloads.html

**Comment by cjbj** [comment]
@Sublime1 I'd recommend updating to the latest 19c instant client because we continue to make Release Updates available for this LTS version, whereas 18 is already out of support.

**Comment by Sublime1** [comment]
@cjbj Thanks for the tip. Is there any reason I shouldn't just go directly to the 21c instant client?

**Comment by cjbj** [comment]
@Sublime1 21c is an innovation release with a shorter lifespan than  the long-term support release 19c. If you adopt 21c be prepared to move to the next, future Instant Client release sooner than if you use 19c.  It's your choice.  The decision is really between how you value long term environment stability and being able to use new features (some of which may need 21c DB).

**Comment by Sublime1** [comment]
@cjbj Thanks again for the clarification. I hadn't been aware of the "innovation release" concept and had just assumed that 21c was the one to go for. That's very good to know.

**Comment by Sublime1** [comment]
Hi @christianshay 
after getting the client I worked, I decide to also try getting the Oracle DB running under Docker too, but this is proving a little more difficult. Would you be able to share the other files that you use in your presentation e.g. db.rsp, orclcdb.bat, post_install.bat, startdb1.sql and startdb2.sql? 

Also, you said to use "Oracle Database 19.7 or later", bit 19.3 is the latest version available to download from https://www.oracle.com/database/technologies/oracle19c-windows-downloads.html Was that a typo as you say 19.3 in your PDF?

Right now I'm blocked on a very strange error where it seems to ignore the value of ORACLE_HOME that I set, and instead sets ORACLE_HOME to the value of ORACLE_BASE. Then it complains: [FATAL] [INS-32052] Oracle base and Oracle home locations are same. 

Would be grateful for any help and especially to see your response file and other scripts.

Thanks in advance!

**Comment by orest-gulman** [comment]
> 

This is mine, is also apply time zone patch and execute few .sql after installation:
`@echo off
cd /d %~dp0
set /p databaseInstanceORADATA="Please specify database instance ORADATA directory (default: %SystemDrive%\oracle\oradata): "
if "%databaseInstanceORADATA%"=="" set databaseInstanceORADATA=%SystemDrive%\oracle\oradata
echo.
set /p databaseInstanceName="Please specify database instance name (maximum 8 symbols, default: ORA193): "
if "%databaseInstanceName%"=="" set databaseInstanceName=ORA193
set databaseInstanceName=%databaseInstanceName:~0,8%
echo.
set /p databaseInstanceSYSDBAPassword="Please specify database instance SYSDBA password (default: password): "
if "%databaseInstanceSYSDBAPassword%"=="" set databaseInstanceSYSDBAPassword=password
echo.
set /p databaseListenerName="Please specify database listener name (default: LISTENER): "
if "%databaseListenerName%"=="" set databaseListenerName=LISTENER
echo.
set /p databaseListenerPort="Please specify database listener port (default: 1521): "
if "%databaseListenerPort%"=="" set databaseListenerPort=1521
echo.
echo Setting Oracle Inventory location...
reg add "HKEY_LOCAL_MACHINE\SOFTWARE\Oracle" /v "inst_loc" /t REG_SZ /d "%ProgramFiles%\Oracle\Inventory" /f
echo.
echo Installing Oracle Database software... This may take a while, do not close this window!
"%SystemDrive%\oracle\product\19.0.0\dbhome_1\setup.exe" -responseFile "%~dp0\Install.rsp" ORACLE_BASE="%SystemDrive%\oracle" ORACLE_HOME="%SystemDrive%\oracle\product\19.0.0\dbhome_1" -silent -waitforcompletion -nowait
echo.
echo Creating directories...
md "%SystemDrive%\oracle\audit"
md "%SystemDrive%\oracle\admin\%databaseInstanceName%\adump"
md "%SystemDrive%\oracle\admin\%databaseInstanceName%\dpdump"
md "%SystemDrive%\oracle\admin\%databaseInstanceName%\pfile"
md "%SystemDrive%\oracle\cfgtoollogs\dbca\%databaseInstanceName%"
md "%databaseInstanceORADATA%\%databaseInstanceName%"
echo.
echo Setting Oracle Base directory permissions...
icacls "%SystemDrive%\oracle" /reset /T /C /Q
icacls "%SystemDrive%\oracle" /grant:r ORA_OraDB19Home1_SVCACCTS:(OI)(CI)M /T /C /Q
if not "%databaseInstanceORADATA%"=="%SystemDrive%\oracle\oradata" (
	icacls "%databaseInstanceORADATA%" /reset /T /C /Q
	icacls "%databaseInstanceORADATA%" /grant:r ORA_OraDB19Home1_SVCACCTS:^(OI^)^(CI^)M /T /C /Q
)
echo Installing database instance... This may take a while, do not close this window!
set ORACLE_SID=%databaseInstanceName%
set ORACLE_HOME=%SystemDrive%\oracle\product\19.0.0\dbhome_1
setx ORACLE_HOME %SystemDrive%\oracle\product\19.0.0\dbhome_1 /m
set PERL5LIB=%ORACLE_HOME%/rdbms/admin;%PERL5LIB%
set PATH=%ORACLE_HOME%\bin;%ORACLE_HOME%\perl\bin;%PATH%
copy /y "C:\scripts\init.ora" "%SystemDrive%\oracle\product\19.0.0\dbhome_1\database"
"%SystemDrive%\oracle\product\19.0.0\dbhome_1\bin\oradim.exe" -new -sid %databaseInstanceName% -startmode auto -srvcstart system
start /wait cmd /c "%SystemDrive%\oracle\product\19.0.0\dbhome_1\bin\sqlplus.exe /nolog @C:\scripts\Install.sql %databaseInstanceSYSDBAPassword%"
echo.
echo Setting %databaseInstanceName% database instance as default ORACLE_SID in Windows Registry...
reg add "HKEY_LOCAL_MACHINE\SOFTWARE\Oracle\KEY_OraDB19Home1" /v "ORACLE_SID" /d "%databaseInstanceName%" /f
echo.
echo Generating Oracle Database network configuration files...
set databaseNetConfigFileExists=false
if exist "%SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\listener.ora" (
	set databaseNetConfigFileExists=true
	rename "%SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\listener.ora" "listener-%date:~0,2%.%date:~3,2%.%date:~6,4%-%time:~0,2%.%time:~3,2%.%time:~6,2%.bak"
	echo Warning! listener.ora file already exists. The file has been backed up and overwritten.
)
@echo # listener.ora Network Configuration File: %SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\listener.ora> "%SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\listener.ora"
@echo # Generated by Oracle configuration tools.>> "%SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\listener.ora"
@echo.>> "%SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\listener.ora"
@echo %databaseListenerName% =>> "%SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\listener.ora"
@echo   (DESCRIPTION =>> "%SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\listener.ora"
@echo     (ADDRESS = (PROTOCOL = TCP)(HOST = 0.0.0.0)(PORT = %databaseListenerPort%))>> "%SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\listener.ora"
@echo   )>> "%SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\listener.ora"
@echo.>> "%SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\listener.ora"
if exist "%SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\tnsnames.ora" (
	set databaseNetConfigFileExists=true
	rename "%SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\tnsnames.ora" "tnsnames-%date:~0,2%.%date:~3,2%.%date:~6,4%-%time:~0,2%.%time:~3,2%.%time:~6,2%.bak"
	echo Warning! tnsnames.ora file already exists. The file has been backed up and overwritten.
)
@echo # tnsnames.ora Network Configuration File: %SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\tnsnames.ora> "%SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\tnsnames.ora"
@echo # Generated by Oracle configuration tools.>> "%SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\tnsnames.ora"
@echo.>> "%SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\tnsnames.ora"
@echo %databaseInstanceName% =>> "%SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\tnsnames.ora"
@echo   (DESCRIPTION =>> "%SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\tnsnames.ora"
@echo     (ADDRESS_LIST =>> "%SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\tnsnames.ora"
@echo       (ADDRESS = (PROTOCOL = TCP)(HOST = localhost)(PORT = %databaseListenerPort%))>> "%SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\tnsnames.ora"
@echo     )>> "%SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\tnsnames.ora"
@echo     (CONNECT_DATA =>> "%SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\tnsnames.ora"
@echo       (SERVER = DEDICATED)>> "%SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\tnsnames.ora"
@echo       (SERVICE_NAME = %databaseInstanceName%)>> "%SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\tnsnames.ora"
@echo     )>> "%SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\tnsnames.ora"
@echo   )>> "%SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\tnsnames.ora"
@echo.>> "%SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\tnsnames.ora"
@echo ALTER SYSTEM SET LOCAL_LISTENER="(ADDRESS = (PROTOCOL = TCP)(HOST = localhost)(PORT = %databaseListenerPort%))";> "C:\scripts\Listener.sql"
powershell -Command "& {((Get-Content -Path "%SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\sqlnet.ora" -Raw) -replace 'NTS','NONE' | Set-Content -Path "%SystemDrive%\oracle\product\19.0.0\dbhome_1\NETWORK\ADMIN\sqlnet.ora") }"
echo.
if "%databaseNetConfigFileExists%"=="true" (
	echo Existing Oracle Database network configuration files have been updated, please check.
	explorer "%SystemDrive%\oracle\product\19.0.0\dbhome_1\network\admin"
	pause
)
echo.
echo Configuring %databaseListenerName% listener on port %databaseListenerPort%...
lsnrctl start %databaseListenerName%
echo.
echo Binding %databaseInstanceName% database instance to %databaseListenerName% listener...
set oracle_sid=%databaseInstanceName%
echo exit | sqlplus sys/%databaseInstanceSYSDBAPassword% as sysdba @"C:\scripts\Listener.sql"
echo exit | sqlplus sys/%databaseInstanceSYSDBAPassword% as sysdba @"C:\scripts\app\password_policy.sql"
echo exit | sqlplus sys/%databaseInstanceSYSDBAPassword% as sysdba @"C:\scripts\app\Create_sysadmin.sql" master password
echo Applying timezone patch...
cd %ORACLE_HOME%\Opatch\32327201
call %ORACLE_HOME%\Opatch\opatch.bat apply -silent
echo exit | sqlplus sys/%databaseInstanceSYSDBAPassword% as sysdba @"C:\scripts\patch\Patch.sql"
echo Configuring services...
sc config "OracleJobScheduler%databaseInstanceName%" start=auto
REM net start "OracleJobScheduler%databaseInstanceName%"
REM net stop "OracleOraDB19Home1TNSListener"
sc config "OracleOraDB19Home1TNSListener" start=auto
REM net start "OracleOraDB19Home1TNSListener"`

**Comment by Sublime1** [comment]
@orest-gulman Thanks very much! This is orclcdb.bat, right? And what version of Oracle DB did you use?
You didn't have the problem I reported with the value of ORACLE_HOME getting ignored?

I would really love to see a working db.rsp file too, as clearly I've done something wrong with mine.

Thanks again!

**Comment by orest-gulman** [comment]
@Sublime1 ver: 19.3
I had similar issue with ORACLE_HOME, I'm my case the Oracle client wasn't working in case of running container using Docker Desktop, at the same time it was working normally using Docker Engine.
I fix it adding line below into installation script:
setx ORACLE_HOME %SystemDrive%\oracle\product\19.0.0\dbhome_1 /m
I've upload all installation files, and in case of issues, please open an case on GitHub and I will look into it:
https://github.com/orest-gulman/Oracle-Database-on-Windows-Container

**Comment by Sublime1** [comment]
@orest-gulman That's very kind of you, many thanks again.
